runtime/cudaq/platform/default/rest/CMakeLists.txt

```diff
@@ -31,7 +31,8 @@ target_link_libraries(cudaq-rest-qpu
     cudaq-operator
     cudaq-common 
   PRIVATE 
-    cudaq-mlir-runtime 
+    cudaq-mlir-compiler
+    nvqir
```

Comment:
Do you really need nvqir here? Or do you just need it when you compile a unit test or executable? If I look in BaseRemoteRESTQPU, I don't think I see any explicit callouts to nvqir symbols. They will be called when we emulate  via simulation, but those symbols need not be here at this time I would think (goes toward my whole thing about decoupling these pieces for better modularity)